tests/testlineargradientbrush.c

```diff
@@ -0,0 +1,1313 @@
+
+    status = GdipGetLineRectI (brush, &lineRect);
+    assert (rect.X == 1);
+    assert (rect.Y == 3);
+    assert (rect.Width == 1);
+    assert (rect.Height == 2);
+
+    status = GdipGetLineRectI (NULL, &lineRect);
+    assert (status == InvalidParameter);
+
+    status = GdipGetLineRectI (brush, NULL);
+    assert (status == InvalidParameter);
+
+    GdipDeleteBrush ((GpBrush *) brush);
+}
+
+static void test_setLineGammaCorrection ()
+{
+    GpStatus status;
+    GpLineGradient *brush;
+    GpRectF rect = { 1, 3, 1, 2 };
+    BOOL useGammaCorrection;
+
+    GdipCreateLineBrushFromRect (&rect, 10, 11, LinearGradientModeHorizontal, WrapModeTile, &brush);
+
+    status = GdipSetLineGammaCorrection (brush, TRUE);
+    assert (status == Ok);
+
+    GdipGetLineGammaCorrection (brush, &useGammaCorrection);
+    assert (useGammaCorrection == TRUE);
+
+    status = GdipSetLineGammaCorrection (NULL, TRUE);
+    assert (status == InvalidParameter);
+
+    GdipDeleteBrush ((GpBrush *) brush);
+}
+
+static void test_getLineGammaCorrection ()
+{
+    GpStatus status;
+    GpLineGradient *brush;
+    GpRectF rect = { 1, 3, 1, 2 };
+    BOOL useGammaCorrection;
+
+    GdipCreateLineBrushFromRect (&rect, 10, 11, LinearGradientModeHorizontal, WrapModeTile, &brush);
+
+    status = GdipGetLineGammaCorrection (brush, &useGammaCorrection);
+    assert (status == Ok);
+    assert (useGammaCorrection == FALSE);
+
+    status = GdipGetLineGammaCorrection (NULL, &useGammaCorrection);
+    assert (status == InvalidParameter);
+
+    status = GdipGetLineGammaCorrection (brush, NULL);
+    assert (status == InvalidParameter);
+
+    GdipDeleteBrush ((GpBrush *) brush);
+}
+
+static void test_getLineBlendCount ()
+{
+    GpStatus status;
+    GpLineGradient *brush;
+    GpRectF rect = { 1, 3, 1, 2 };
+    INT blendCount;
+
+    GdipCreateLineBrushFromRect (&rect, 10, 11, LinearGradientModeHorizontal, WrapModeTile, &brush);
+
+    status = GdipGetLineBlendCount (brush, &blendCount);
+    assert (status == Ok);
+    assert (blendCount == 1);
+
+    status = GdipGetLineBlendCount (NULL, &blendCount);
+    assert (status == InvalidParameter);
+
+    status = GdipGetLineBlendCount (brush, NULL);
+    assert (status == InvalidParameter);
+
+    GdipDeleteBrush ((GpBrush *) brush);
+}
+
+static void test_getLineBlend ()
+{
+    GpStatus status;
+    GpLineGradient *brush;
+    GpRectF rect = { 1, 3, 1, 2 };
+    REAL blend[2];
+    REAL positions[2];
+    REAL largeBlend[3] = { 1, 2, 3 };
+    REAL largePositions[3] = { 0, 0.5, 1 };
+
+    GdipCreateLineBrushFromRect (&rect, 10, 11, LinearGradientModeHorizontal, WrapModeTile, &brush);
+
+    status = GdipGetLineBlend (brush, blend, positions, 1);
+    assert (status == Ok);
+    assert (floatsEqual (blend[0], 1));
+    // FIXME: it appears that Windows 10+ versions of GDI+ don't copy anything to positions.
+    // assert (floatsEqual (positions[0], 0));
```

Comment:
/cc @mellinoe @akoeplinger 

This passes online with rextester.com/l/cpp_online_compiler_visual (an older version of Windows), but fails on Windows 10.

This is a regression in GDI+

Reply:
Hmm, I have no idea where to file a Windows bug, @mellinoe ? 😄